Content.Server/Traits/TraitSystem.Functions.cs

```diff
@@ -444,85 +448,29 @@ public sealed partial class TraitModifyMobState : TraitFunction
-    public bool? AllowTalkingWhileDead;
-
-    [DataField, AlwaysPushInheritance]
-    public bool? DownWhenCrit;
-
-    [DataField, AlwaysPushInheritance]
-    public bool? DownWhenSoftCrit;
-
-    [DataField, AlwaysPushInheritance]
-    public bool? DownWhenDead;
-
-    [DataField, AlwaysPushInheritance]
-    public bool? AllowHandInteractWhileCrit;
-
-    [DataField, AlwaysPushInheritance]
-    public bool? AllowHandInteractWhileSoftCrit;
-
-    [DataField, AlwaysPushInheritance]
-    public bool? AllowHandInteractWhileDead;
+    [DataField(required: true)]
+    public Dictionary<string, ProtoId<MobStateParametersPrototype>> Params = default!;
 
     public override void OnPlayerSpawn(EntityUid uid,
         IComponentFactory factory,
         IEntityManager entityManager,
         ISerializationManager serializationManager)
     {
-        if (!entityManager.TryGetComponent<MobStateComponent>(uid, out var mobStateComponent))
+        if (!entityManager.TryGetComponent<MobStateComponent>(uid, out var comp))
             return;
 
-        if (AllowMovementWhileCrit is not null)
-            mobStateComponent.AllowMovementWhileCrit = AllowMovementWhileCrit.Value;
-
-        if (AllowMovementWhileSoftCrit is not null)
-            mobStateComponent.AllowHandInteractWhileSoftCrit = AllowMovementWhileSoftCrit.Value;
-
-        if (AllowMovementWhileDead is not null)
-            mobStateComponent.AllowMovementWhileDead = AllowMovementWhileDead.Value;
-
-        if (AllowTalkingWhileCrit is not null)
-            mobStateComponent.AllowTalkingWhileCrit = AllowTalkingWhileCrit.Value;
-
-        if (AllowTalkingWhileSoftCrit is not null)
-            mobStateComponent.AllowTalkingWhileSoftCrit = AllowTalkingWhileSoftCrit.Value;
-
-        if (AllowTalkingWhileDead is not null)
-            mobStateComponent.AllowTalkingWhileDead = AllowTalkingWhileDead.Value;
+        var _reflection = IoCManager.Resolve<IReflectionManager>();
+        var _proto = IoCManager.Resolve<IPrototypeManager>();
 
-        if (DownWhenCrit is not null)
-            mobStateComponent.DownWhenCrit = DownWhenCrit.Value;
-
-        if (DownWhenSoftCrit is not null)
-            mobStateComponent.DownWhenSoftCrit = DownWhenSoftCrit.Value;
-
-        if (DownWhenDead is not null)
-            mobStateComponent.DownWhenDead = DownWhenDead.Value;
-
-        if (AllowHandInteractWhileCrit is not null)
-            mobStateComponent.AllowHandInteractWhileCrit = AllowHandInteractWhileCrit.Value;
-
-        if (AllowHandInteractWhileSoftCrit is not null)
-            mobStateComponent.AllowHandInteractWhileSoftCrit = AllowHandInteractWhileSoftCrit.Value;
-
-        if (AllowHandInteractWhileDead is not null)
-            mobStateComponent.AllowHandInteractWhileDead = AllowHandInteractWhileDead.Value;
+        foreach (var pair in Params) {
+            DebugTools.Assert(_reflection.TryParseEnumReference($"enum.MobState.{pair.Key}", out var e), $"MobState.{pair.Key} does not exist.");
```

Comment:
```suggestion
            DebugTools.Assert(_reflection.TryParseEnumReference($"enum.MobState.{pair.Key}", out var e), $"MobState.{pair.Key} does not exist.");
```
You're not allowed to use e as a variable here because it straight up won't exist in Release configuration. Separate out TryParseEnumReference into an If Not {Assert; Continue;} statement to clear the error

Reply:
i am not having any errors here, but sure

Reply:
the null checker actually stops seeing that e won't be null on the line 468, where it's cast into MobState var